braindecode/models/eegitnet.py

```diff
@@ -308,27 +331,22 @@ def __init__(
         self.add_module(
             "dim_reduction",
             nn.Sequential(
-                nn.Conv2d(14, 28, kernel_size=(1, 1)),
-                nn.BatchNorm2d(28),
+                nn.Conv2d(tcn_in_channel, int(tcn_in_channel * 2), kernel_size=(1, 1)),
+                nn.BatchNorm2d(int(tcn_in_channel * 2)),
                 activation(),
-                nn.AvgPool2d((1, 4)),
+                nn.AvgPool2d((1, tcn_kernel_size)),
                 nn.Dropout(drop_prob),
             ),
         )
         # ============== Classifier ==================
         # Moved flatten to another layer
         self.add_module("flatten", nn.Flatten())
 
-        # Incorporating classification module and subsequent ones in one final layer
-        module = nn.Sequential()
-
-        module.add_module(
-            "clf", nn.Linear(int(int(self.n_times / 4) / 4) * 28, self.n_outputs)
+        num_features = int(int(self.n_times / tcn_kernel_size) / 4) * int(
+            tcn_in_channel * 2
```

Comment:
Are you sure about the math?
 
The safe way would be to use:
```suggestion
        num_features = self.get_output_shape()
```
But it would make the model initialisation a bit longer...